Todo.WillowBear/Todo.WillowBear/Data/TodoDbContext.cs

```diff
@@ -0,0 +1,160 @@
+
+            if (todoItem == null)
+            {
+                Log.Debug($"{todoItem} is null");
+            }
+
+
+            try
+            {
+                TodoItems.Add(todoItem!);
+                SaveChanges();
+                Log.Information($"Added {todoItem} to database successfully");
+            }
+            catch (Exception e)
+            {
+
+                Log.Information($"Unable to add new Todo Item. ");
+                Log.Debug(e.Message);
+            }
+        }
+
+        // edit todoitem
+        /// <summary>
+        /// Edits TodoItem in Database
+        /// </summary>
+        /// <param name="todoItem"></param>
+        public void EditTodoItem(TodoItem todoItem)
+        {
+            if (todoItem == null)
+            {
+                Log.Debug($"{todoItem} is null");
+            }
+
+            try
+            {
+                TodoItems.Update(todoItem!);
+                SaveChanges();
+                Log.Information($"Todo Item {todoItem!.Id} has been updated successfully");
+            }
+            catch (Exception e)
+            {
+
+                Log.Information($"Unable to edit Todo Item. ");
+                Log.Debug(e.Message);
+            }
+        }
+
+        // delete todoitem
+        /// <summary>
+        /// Deletes TodoItem from Database
+        /// </summary>
+        /// <param name="todoItem"></param>
+        public void DeleteTodoItem(TodoItem todoItem)
+        {
+            if (todoItem == null)
+            {
+                Log.Debug($"{todoItem} is null");
+            }
+
+            try
+            {
+                TodoItems.Remove(todoItem!);
+                SaveChanges();
+                Log.Information($"Todo Item {todoItem!.Id} has been deleted successfully");
+            }
+            catch (Exception e)
+            {
+                Log.Information($"Unable to delete Todo Item. ");
+                Log.Debug(e.Message);
+            }
+        }
+
+        // get todoitem
+        /// <summary>
+        /// Returns a single TodoItem from Database if it exists
+        /// </summary>
+        /// <param name="id"></param>
+        /// <returns>TodoItem Object || null if does not exist</returns>
+        public TodoItem? GetTodoItemById(int id)
+        {
+            try
+            {
+                var todoItem = TodoItems.Find(id);
+                if (todoItem == null)
+                {
+                    Log.Debug($"Todo Item {id} does not exist");
+                    return null;
+                }
+                else
+                {
+                    Log.Information($"Todo Item {id} has been found");
+                    return todoItem;
+                }
+            }
+            catch (Exception e)
+            {
+                Log.Information($"Unable to retrieve Todo Item. ");
+                Log.Debug(e.Message);
+                return null;
+            }
+        }
+
+        // get all todoitems
```

Comment:
It's ok to add these comments for personal reference but your method signature should be self-explanatory. In most companies, you wouldn't push these types of comments to a team.

Reply:
Fixed.

Can you advise on a good resource or explain what comments should be left for production commits?